connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/outbound/ConnectorJobHandler.java

```diff
@@ -179,12 +192,24 @@ public void handle(final JobClient client, final ActivatedJob job) {
               job.getCustomHeaders().get(Keywords.RESULT_VARIABLE_KEYWORD),
               job.getCustomHeaders().get(Keywords.RESULT_EXPRESSION_KEYWORD));
       result = new ConnectorResult.SuccessResult(response, responseVariables);
+    } catch (ConnectorRetryException ex) {
+      LOGGER.debug(
+          "ConnectorRetryException while processing job: {} for tenant: {}",
+          job.getKey(),
+          job.getTenantId(),
+          ex);
+      String errorCode = ex.getErrorCode();
+      result = handleSDKException(job, ex, ex.getRetryPolicy(), errorCode, retryBackoff);
     } catch (Exception ex) {
       LOGGER.debug(
           "Exception while processing job: {} for tenant: {}", job.getKey(), job.getTenantId(), ex);
       result =
-          new ConnectorResult.ErrorResult(
-              Map.of("error", exceptionToMap(ex)), ex, job.getRetries() - 1, retryBackoff);
+          handleSDKException(
```

Comment:
We handle other exceptions the same way as the `ConnectorRetryException` because there's a retry and a backoff involved anyway (coming from the job itself). 
The only difference is that we won't have any `errorCode`, hence the `CATCH_ALL_ERROR_CODE`.
So it makes sense to re-use the code here for clarity purposes.